Mathlib/Order/SuccPred/Nat.lean

```diff
@@ -0,0 +1,24 @@
+/-
+Copyright (c) 2024 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import Mathlib.Order.SuccPred.LinearLocallyFinite
+import Mathlib.Order.Interval.Finset.Nat
+
+/-!
+# The order-theoretic successor function on `Nat`
+-/
+
+namespace Nat
+
+/-- The order-theoretic successor function on `Nat` is what it should be. -/
+lemma orderSucc_eq_succ (n : ℕ) :
+    Order.succ n = n + 1 := by
```

Comment:
This is already `Order.succ_eq_add_one`

Reply:
I tried to use that in my code, but I'm getting a weird error. There seems to be an instance diamond involving `SuccOrder` instances on `ℕ`. I reported it on Zulip [here](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/Non-defeq.20SuccOrder.20instances)

Reply:
Now resolved using the fixes in #20235.